pkg/crawl/identify.go

```diff
@@ -222,7 +222,11 @@ func (i *Log4jIdentifier) vulnerabilityFileWalkFunc(depth uint, result *Finding,
 			}
 		}
 
-		if strings.HasSuffix(path, ".class") {
+		filename, lastSlash := path, strings.LastIndex(path, "/")
+		if lastSlash != -1 && lastSlash != len(filename)-1 {
+			filename = filename[lastSlash+1:]
+		}
+		if strings.HasSuffix(filename, ".class") {
```

Comment:
Instead of adding another if statement to this block.

Let's match here on `strings.HasSuffix(filename, ".class")` or `strings.HasPrefix(filename, "JndiManager.")`,
then pass the filename to `lookForClassFileMatch` where we can hash it based on the filename having a  `"JndiManager."` prefix at https://github.com/palantir/log4j-sniffer/blob/4002781d2f6b258bc2bc043d1452d809fde8c96c/pkg/crawl/identify.go#L296

This will keep the class file matching logic more contained.